sdk/src/Services/S3/Custom/Transfer/Internal/IDownloadManager.cs

```diff
@@ -46,8 +46,9 @@ internal interface IDownloadManager : IDisposable
         /// </summary>
         /// <param name="discoveryResult">Results from the discovery phase.</param>
         /// <param name="cancellationToken">A token to cancel the download operation.</param>
+        /// <param name="progressCallback">Optional callback for progress tracking events.</param>
         /// <returns>A task that completes when all downloads finish or an error occurs.</returns>
-        Task StartDownloadsAsync(DownloadDiscoveryResult discoveryResult, CancellationToken cancellationToken);
+        Task StartDownloadsAsync(DownloadDiscoveryResult discoveryResult, CancellationToken cancellationToken, EventHandler<WriteObjectProgressArgs> progressCallback = null);
```

Comment:
It is the strong convention that `CancellationToken` should be the last argument and I would avoid putting a default on `progressCallback` because that can cause the caller to forget to pass along the callback.

Reply:
updated the order in https://github.com/aws/aws-sdk-net/pull/4139/commits/7a4a1e1f9313958f83f225291f31c81704c19066